VPLAY-10805 class sep and main_aamp dep untangling

pstroffolino opened the pull request:
VPLAY-10805 class sep and main_aamp dep untangling

Reason for Change: teh debt

pstroffolino committed:
VPLAY-10805 class sep and main_aamp dep untangling

Reason for Change: teh debt

Signed-off-by: Philip Stroffolino <philip_stroffolino@cable.comcast.com>

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
f6eeb33ad2f6f3c3baf50a4663f07788
PRESENT
-->

# Coverity Issue - Data race condition
A wait is performed without a loop. If there is a spurious wakeup, the condition may not be satisfied.

_Medium Impact, CWE-none_
_BAD_CHECK_OF_WAIT_COND_

## How to fix
 Check the wait condition in a loop, with the lock held. The lock must not be released between the condition and the wait.

## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[priv_aamp.cpp:7537](undefined//blob/undefined/priv_aamp.cpp#L7537)

pstroffolino committed:
fix for tsb utests

Signed-off-by: Philip Stroffolino <philip_stroffolino@cable.comcast.com>

pstroffolino committed:
Reason for Change: new split out headers AAMPAnomalyMessageType.h, AuthTokenErrors.h, AudioTrackInfo.h, TextTrackInfo.h, VideoZoomMode
Removed dependency on aamp public headers from tsprocessor, AampGrowableBuffer

Signed-off-by: Philip Stroffolino <philip_stroffolino@cable.comcast.com>

pstroffolino committed:
whitespace and aamp include headers

Signed-off-by: Philip Stroffolino <philip_stroffolino@cable.comcast.com>

pstroffolino committed:
Merge branch 'dev_sprint_25_2' into feature/VPLAY-10805-class-seperation
